Sources/Regex/Anchor.swift

```diff
@@ -0,0 +1,18 @@
+/// An token that matches against a certain position in the subject.
+public enum Anchor: String, Hashable {
+  case lineStart = "^"
+  case lineEnd = "$"
+  case wordBoundary = "\\b"
+  case nonWordBoundary = "\\B"
```

Comment:
I'm fine with more broadly calling built-in assertions "anchors", which is what `.NET` calls the IIRC

Reply:
Yeah I was going off what Oniguruma calls them